pandas/core/categorical.py

```diff
@@ -1,2331 +1,2 @@
-                          get_option("display.max_categories"))
-        from pandas.io.formats import format as fmt
-        if len(self.categories) > max_categories:
-            num = max_categories // 2
-            head = fmt.format_array(self.categories[:num], None)
-            tail = fmt.format_array(self.categories[-num:], None)
-            category_strs = head + ["..."] + tail
-        else:
-            category_strs = fmt.format_array(self.categories, None)
-
-        # Strip all leading spaces, which format_array adds for columns...
-        category_strs = [x.strip() for x in category_strs]
-        return category_strs
-
-    def _repr_categories_info(self):
-        """ Returns a string representation of the footer."""
-
-        category_strs = self._repr_categories()
-        dtype = getattr(self.categories, 'dtype_str',
-                        str(self.categories.dtype))
-
-        levheader = "Categories ({length}, {dtype}): ".format(
-            length=len(self.categories), dtype=dtype)
-        width, height = get_terminal_size()
-        max_width = get_option("display.width") or width
-        if com.in_ipython_frontend():
-            # 0 = no breaks
-            max_width = 0
-        levstring = ""
-        start = True
-        cur_col_len = len(levheader)  # header
-        sep_len, sep = (3, " < ") if self.ordered else (2, ", ")
-        linesep = sep.rstrip() + "\n"  # remove whitespace
-        for val in category_strs:
-            if max_width != 0 and cur_col_len + sep_len + len(val) > max_width:
-                levstring += linesep + (" " * (len(levheader) + 1))
-                cur_col_len = len(levheader) + 1  # header + a whitespace
-            elif not start:
-                levstring += sep
-                cur_col_len += len(val)
-            levstring += val
-            start = False
-        # replace to simple save space by
-        return levheader + "[" + levstring.replace(" < ... < ", " ... ") + "]"
-
-    def _repr_footer(self):
-
-        return u('Length: {length}\n{info}').format(
-            length=len(self), info=self._repr_categories_info())
-
-    def _get_repr(self, length=True, na_rep='NaN', footer=True):
-        from pandas.io.formats import format as fmt
-        formatter = fmt.CategoricalFormatter(self, length=length,
-                                             na_rep=na_rep, footer=footer)
-        result = formatter.to_string()
-        return compat.text_type(result)
-
-    def __unicode__(self):
-        """ Unicode representation. """
-        _maxlen = 10
-        if len(self._codes) > _maxlen:
-            result = self._tidy_repr(_maxlen)
-        elif len(self._codes) > 0:
-            result = self._get_repr(length=len(self) > _maxlen)
-        else:
-            msg = self._get_repr(length=False, footer=True).replace("\n", ", ")
-            result = ('[], {repr_msg}'.format(repr_msg=msg))
-
-        return re
```

Comment:
Is this unavoidable for older pickle files?

Reply:
not sure what you are pointing to. but you need to have an entry in ``pandas.compat/pickle_compat.py`` to handle the pickle compat

Reply:
Thanks. I thought we had something like that.